packages/consumption/src/modules/attributes/local/LocalAttribute.ts

```diff
@@ -125,10 +125,6 @@ export class LocalAttribute extends CoreSynchronizable implements ILocalAttribut
             return isOwnSharedAttribute;
         }
 
-        if (this.isIdentityAttribute()) {
```

Comment:
I adjusted OwnSharedIdentityAttribute and also OwnSharedRelationshipAttribute, since in the case of own shared third party RelationshipAttributes the `shareInfo.sourceAttribute` property may be defined.

Reply:
Gotcha. I also noticed that the type `RepositoryAttribute` is incorrectly defined:

```ts
export type RepositoryAttribute = IdentityAttribute & { shareInfo: undefined };
```

This should rather be

```ts
export type RepositoryAttribute = LocalAttribute & { content: IdentityAttribute } & { shareInfo: undefined };
```

Right?

Reply:
It is! But this can also be done "inline".

```

export type RepositoryAttribute = LocalAttribute & { content: IdentityAttribute, shareInfo: undefined };
```

Reply:
I see the other types can also be simplified, all those `& {...}` are kind of hard to process.

Reply:
@slavistan & @Milena-Czierlinski as this could lead to errors I have done this in another branch to main: https://github.com/nmshd/runtime/pull/67